receiver/k8sclusterreceiver/internal/collection/nodes.go

```diff
@@ -83,6 +113,15 @@ var nodeConditionValues = map[corev1.ConditionStatus]int64{
 	corev1.ConditionUnknown: -1,
 }
 
+func nodeAllocatableValue(node *corev1.Node, allocatType corev1.ResourceName) (int64, bool) {
+	value, ok := node.Status.Allocatable[allocatType]
+	if !ok {
+		return 0, false
+	}
+	val, _ := value.AsInt64()
```

Comment:
Yes, 0 is returned in case of an error and the value of `_` will be false. If we return false from this function then the respective metric will be skipped.

Reply:
Why should we record `0` if error returned here? Skipping the metric seems to be right way to handle it. I would recommend returning (int64, error) instead of (int64, bool). Also I would add a debug log message on error returned by `nodeAllocatableValue`

Reply:
Here in case of error we will get (0,false) as the function `AsInt64` will return (int64,bool).
In case we get (0,false) we can put a debug statement like `metric 'x' has value 'y' which cannot be converted to int64` and skip that metric by returning (0,false).
@dmitryax What are your views on this?

Reply:
Sounds good. And another error should be returned if we fail to get a value from `node.Status.Allocatable` map

Reply:
Added debug statements.